hoomd/md/constrain.py

```diff
@@ -342,85 +176,69 @@ class rigid(ConstraintForce):
+            particles
+        orientations (list): List of orientations of constituent particles
+            (**optional**)
+        charge (list): List of charges of constituent particles
+            (**optional**)
+        diameters (list): List of diameters of constituent particles
+            (**optional**)
 
-        # initialize the base class
-        ConstraintForce.__init__(self)
+    .. caution::
+        The constituent particle type must be exist.
+        If it does not exist, it can be created on the fly using
+        ``system.particles.types.add('A_const')``.
 
-        self.composite = True
+    Example::
 
-        # create the c++ mirror class
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            self.cpp_force = _md.ForceComposite(
-                hoomd.context.current.system_definition
+        rigid = constrain.rigid()
+        rigid.set_param(
+            'A',
+            types = ['A_const', 'A_const'],
+            positions = [(0,0,1),(0,0,-1)]
             )
-        else:
-            self.cpp_force = _md.ForceCompositeGPU(
-                hoomd.context.current.system_definition
+        rigid.set_param(
+            'B',
+            types = ['B_const', 'B_const'],
+            positions = [(0,0,.5),(0,0,-.5)]
             )
 
-        hoomd.context.current.system.addCompute(self.cpp_force, self.force_name)
-
-    def set_param(
-        self,
-        type_name,
-        types,
-        positions,
-        orientations=None,
-        charges=None,
-        diameters=None,
-    ):
-        R"""Set constituent particle types and coordinates for a rigid body.
-
-        Args:
-            type_name (str): The type of the central particle
-            types (list): List of types of constituent particles
-            positions (list): List of relative positions of constituent
-                particles
-            orientations (list): List of orientations of constituent particles
-                (**optional**)
-            charge (list): List of charges of constituent particles
-                (**optional**)
-            diameters (list): List of diameters of constituent particles
-                (**optional**)
-
-        .. caution::
-            The constituent particle type must be exist.
-            If it does not exist, it can be created on the fly using
-            ``system.particles.types.add('A_const')``.
-
-        Example::
+    """
 
-            rigid = constrain.rigid()
-            rigid.set_param(
-                'A',
-                types = ['A_const', 'A_const'],
-                positions = [(0,0,1),(0,0,-1)]
-                )
-            rigid.set_param(
-                'B',
-                types = ['B_const', 'B_const'],
-                positions = [(0,0,.5),(0,0,-.5)]
-                )
+    _cpp_class_name = "ForceComposite"
+    def __init__(self):
+        pass
 
-        """
+    def set_params(
```

Comment:
Let's add documentation for this method.

Reply:
I think this method is going to be replaced.